### ISSUE TITLE
Reaper becomes unresponsive when typing into search edit field using JAWS screen reader

### ISSUE THREAD

**Issue by ptorpey** [surge-synthesizer/surge#7881]
This is an accessibility issue for people using a screen reader.

I am using Reaper and the latest version of Surge XT just downloaded today.

I am unable to type in the edit field to search for instruments using the JAWS sdreen reader, but using the NVDA screen reader works fine.

Here is how the problem occurs:
1.	 Open the FX dialog with Surge XT on a track by hitting “F”
2.	Hit F6
3.	Hit control+F.  This brings up a list of instruments.
4.	If I type a new instrument or search item into this edit or list view with JAWS, Reaper hangs and my whole system hangs.  If I restart JAWS and try to alt+F4 out of Reaper, I am told that Reaper is not responding and asked if I want to close the program.
5.	If I type a new instrument or search item in the edit / list field using NVDA, a new list populates with instruments that match the entered text.

I am using the latest version of Reaper and JAWS 2025.

--Pete

**Bug Description:**

**Surge XT Version**
This information is found on the About screen, which you get to from the bottom right menu.

* Version:
* Plugin Type: [VST2, VST3, AU]
* Bitness: [32-bit or 64-bit]

**Reproduction Steps:**
2. Click on '...'
3. Scroll down to '...'

**Expected Behavior:**

**Screenshots:**
If applicable, add screenshots/GIF/videos to help explain your problem.

**Computer Information (please complete the following!):**
 - OS: [e.g. Windows 10]
 - Host: [e.g. Bitwig, Logic, Cubase, Reaper]
 - Version: [e.g. 2.4]

**Additional Information:**

**Comment by ptorpey** [comment]
[PT] I neglected to specify the version of Surge XT.  I downloaded the file:

surge-xt-win64-1.3.4-setup.exe

 

--Pete

**Comment by ptorpey** [comment]
Just to follow up on this, a JAWS developer figured out what was going on with this issue.

 

The problem turned out not to be a JAWS problem but something wrong in the Surge code.  Here is the comment from the JAWS developer on what was wrong and how to fix the issue in Surge:

 

“Basically, the UIA ITextRangeProvider method Move indicates that movement is successful indefinitely, when we are walking backward through a range to find the start.

i.e. range->Move(TextUnit_Character, -1, &unitsMoved);

always returns success and unitsMoved=1 when it should either return S_FALSE or set unitsMoved to 0 once the start of the edit range is reached.

I don’t know if it always does this or only on certain conditions such as when you initially start typing, but, we were caught in an infinite loop.”

 

Hope that helps.

 

--Pete

**Comment by baconpaul** [comment]
Wow that sounds like a juce bug to me. We don’t interact directly with those apis in surge
Let me figure out how to get this report to them

since 134 we’ve upgraded to juce 8 so I do wonder if the nightly exhibits this behavior also

**Comment by ptorpey** [comment]
Would you like me to test one of the nightly builds?  Would that be helpful?

 

Just let me know.

 

Thanks for passing this along to the Juice team.  I have been told that JAWS will have a fix for this in the next beta and next publicly released version.

 

--Pete

**Comment by baconpaul** [comment]
If you could that would be great.

https://surge-synthesizer.github.io/nightly_XT/

**Comment by ptorpey** [comment]
I downloaded the nightly release:

surge-xt-win64-NIGHTLY-2024-12-03-3dc58a3-setup.exe

and have some confusing results.

 

When I load up the VSTi Surge XT version, hit F6, and then control+F, the search field locks up as before.

 

When I do the same thing with the CLAPi version, hitting control+F brings up a search field with all of the FX on my system, not any instruments.  There is a CLAP Surge XT FX item that I can also add which brings up the usual dialog for Surge FX’s.  Perhaps the CLAPi version just opens the wrong database when using search.

 

???

 

--Pete

**Comment by ptorpey** [comment]
Paul,

 

Just FYI, this issue has been resolved in the latest JAWS beta.  I believe the release version with this fix should be available around March, 2025.

 

Still, would be good to follow up with the Juice team to see if they can fix the issue on their end in case it also affects any other Juice implementations.  

 

Thanks again for following up.  Happy new year!

 

--Pete

**Comment by baconpaul** [comment]
Oh great!

I'll forward it to that info to the lead accessibility dev there. Thank you.

**Comment by mkruselj** [comment]
Shall we close this one then?

**Comment by baconpaul** [comment]
yes i think so. I forwarded it to juce team.

**Comment by ptorpey** [comment]
Thanks again for following up and passing this along to the Juice team.

 

--Pete
